plenum/test/req_drop/test_req_drop_on_commit_phase_primary.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+
+from plenum.test.helper import sdk_send_random_requests
+from stp_core.loop.eventually import eventually
+from plenum.common.messages.node_messages import Commit
+from plenum.test.delayers import delay
+from plenum.test.propagate.helper import recvdRequest, recvdPropagate, \
+    sentPropagate, recvdPrepareForInstId, recvdCommitForInstId
+from plenum.test.test_node import TestNode
+from plenum.test.node_request.helper import sdk_ensure_pool_functional
+
+
+howlong = 20
+initial_ledger_size = 0
+
+
+@pytest.fixture(scope="module")
+def tconf(tconf):
+    OUTDATED_REQS_CHECK_ENABLED_OLD = tconf.OUTDATED_REQS_CHECK_ENABLED
+    OUTDATED_REQS_CHECK_INTERVAL_OLD = tconf.OUTDATED_REQS_CHECK_INTERVAL
+    PROPAGATES_PHASE_REQ_TIMEOUT_OLD = tconf.PROPAGATES_PHASE_REQ_TIMEOUT
+    ORDERING_PHASE_REQ_TIMEOUT_OLD = tconf.ORDERING_PHASE_REQ_TIMEOUT
+
+    tconf.OUTDATED_REQS_CHECK_ENABLED = True
+    tconf.OUTDATED_REQS_CHECK_INTERVAL = 1
+    tconf.PROPAGATES_PHASE_REQ_TIMEOUT = 3600
+    tconf.ORDERING_PHASE_REQ_TIMEOUT = 10
+    yield tconf
+
+    tconf.OUTDATED_REQS_CHECK_ENABLED = OUTDATED_REQS_CHECK_ENABLED_OLD
+    tconf.OUTDATED_REQS_CHECK_INTERVAL = OUTDATED_REQS_CHECK_INTERVAL_OLD
+    tconf.PROPAGATES_PHASE_REQ_TIMEOUT = PROPAGATES_PHASE_REQ_TIMEOUT_OLD
+    tconf.ORDERING_PHASE_REQ_TIMEOUT = ORDERING_PHASE_REQ_TIMEOUT_OLD
+
+
+@pytest.fixture()
+def setup(txnPoolNodeSet, looper, sdk_pool_handle, sdk_wallet_client):
+    global initial_ledger_size
+    A, B, C, D = txnPoolNodeSet  # type: TestNode
+    lagged_node = A
+    frm = [B, C, D]
+    delay(Commit, frm=frm, to=lagged_node, howlong=howlong)
+    initial_ledger_size = lagged_node.domainLedger.size
+    request_couple_json = sdk_send_random_requests(
+        looper, sdk_pool_handle, sdk_wallet_client, 1)
+    return request_couple_json
+
+
+# NOTE: if this test fails intermittently then ORDERING_PHASE_REQ_TIMEOUT
+# and Prepares and Commits wait timeouts should be tuned as we can not
+# synchronize receiving of Prepares and dropping of the request, for now
+# it is controlled just using timeouts so that we can drop the request
+# before all Prepares received.
+def test_req_drop_on_commit_phase_on_master_primary_and_then_ordered(
+        tconf, setup, looper, txnPoolNodeSet,
```

Comment:
This test (with Primary) looks pretty similar to the previous test (without primary).
So, it may make sense to parametrize it and have just one test.

Reply:
Yes, it's true, but unfortunately the parametrisation of tests makes checks of number of received messages very unclear as the second run is done relatively to the first run. Also it is very hard to read and very hard to support and make changes, so in this particular situation the code duplication is better solution than the parametrisation of tests.